File: 4. Android FrameWork with JAVA/Aug 28/My Demo/app/src/main/java/com/example/mydemo/MainActivity.java
package com.example.mydemo;

import android.os.Bundle;

import com.google.android.material.snackbar.Snackbar;

import androidx.appcompat.app.AppCompatActivity;

import android.view.View;

import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

import com.example.mydemo.databinding.ActivityMainBinding;

import android.view.Menu;
import android.view.MenuItem;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private EditText courseName, courseTracks, courseDuration, courseDesc;
    private Button addcourse;
    private  DbHandler dbHandler;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        courseName = findViewById(R.id.coursename);
        courseTracks = findViewById(R.id.courseTracks);
        courseDuration = findViewById(R.id.courseDuration);
        courseDesc = findViewById(R.id.coursedesc);
        addcourse = findViewById(R.id.addcourse);
        dbHandler = new DbHandler(MainActivity.this);
        addcourse.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String cName = courseName.getText().toString();
                String cTracks = courseTracks.getText().toString();
                String cDesc = courseDesc.getText().toString();
                String cDuration = courseDuration.getText().toString();
                // validation
                if(cName.isEmpty() && cDesc.isEmpty() && cDuration.isEmpty() && cTracks.isEmpty()) {
                    Toast.makeText(MainActivity.this, "Please Enter all the data", Toast.LENGTH_SHORT).show();
                    return;
                }
                dbHandler.addNewCourse(cName, cDuration, cDesc, cTracks);
                Toast.makeText(MainActivity.this, "Course Added Successfully", Toast.LENGTH_SHORT).show();
            }
        });
    }

}

File: 6. Android FrameWork with Kotlin/Aug 1/Kotlin Demo/app/src/main/java/com/example/kotlindemo/MainActivity.kt
package com.example.kotlindemo

import android.os.Bundle
import android.widget.Button
import android.widget.Toast
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import com.example.kotlindemo.ui.theme.KotlindemoTheme

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main_activity)

        val buttonWelcome: Button = findViewById(R.id.buttonWelcome)

        buttonWelcome.setOnClickListener {
            Toast.makeText(this, "Hello, Welcome Amulya", Toast.LENGTH_SHORT).show()
        }
    }
}